dbms/src/Storages/KVStore/MultiRaft/ApplySnapshot.cpp

```diff
@@ -351,7 +351,7 @@ void KVStore::onSnapshot(
 
         tmt.getRegionTable().shrinkRegionRange(*new_region);
     }
-
+    new_region->updateSnapshotAppliedTime();
```

Comment:
We always create a `new_region` instance to apply snapshot instead of modifying the `old_region` in-place. So I think the `tiflash_raft_long_term_event_duration_seconds, type_apply_snapshot_gap` is not get reported at all?

Reply:
Yes, I have tried to fix this by fetch the `lastSNapshotAppliedTime()` from the old_region(if any) to compute.